cmake/Modules/GrPython.cmake

```diff
@@ -37,9 +37,14 @@ set(PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE} CACHE FILEPATH "python interpreter")
 set(QA_PYTHON_EXECUTABLE ${QA_PYTHON_EXECUTABLE} CACHE FILEPATH "python interpreter for QA tests")
 
 add_library(Python::Python INTERFACE IMPORTED)
+if(APPLE)
+    set_target_properties(Python::Python PROPERTIES
+      INTERFACE_INCLUDE_DIRECTORIES "${PYTHON_INCLUDE_DIRS}"
+      INTERFACE_LINK_LIBRARIES "-undefined dynamic_lookup"
+      )
 # Need to handle special cases where both debug and release
 # libraries are available (in form of debug;A;optimized;B) in PYTHON_LIBRARIES
-if(PYTHON_LIBRARY_DEBUG AND PYTHON_LIBRARY_RELEASE)
+elseif(PYTHON_LIBRARY_DEBUG AND PYTHON_LIBRARY_RELEASE)
```

Comment:
Is it not possible to be both on Mac, and also have PYTHON_LIBRARY_DEBUG?

Reply:
yes this is possible on macOS ... but the linkage depends on how Python ABI is provided. If in a framework then we link against the master framework & one can select debug or not via special environment variables. If as libraries, then we link against the desired library ... same as other Linux & others. One can always link against the object file, of course, but with frameworks there's a more robust alternative.

Reply:
This change is, really, about as good as it gets for macOS with respect to allowing future Python linkage to work with current ABI, because of how the macOS DYLD works. We can fixup the debug + release case later if desired.